src/main/setLanguage.js

```diff
@@ -0,0 +1,67 @@
+import GettextJS from 'gettext.js';
+import gettextParser from 'gettext-parser';
+import { ipcMain } from 'electron';
+import fs from 'fs';
+import { ipcMainChannels } from './ipcMainChannels';
+
+const i18n = new GettextJS();
+
+/** Read PO file into gettext.js formatted message catalog object. */
+function readMessageCatalog(messageCatalogPath) {
+  const input = fs.readFileSync(messageCatalogPath);
+  // parse PO file into an object in gettext-parser format
+  // https://github.com/smhg/gettext-parser#data-structure-of-parsed-mopo-files
+  const rawPO = gettextParser.po.parse(input);
+
+  // convert from gettext-parser format to gettext.js format
+  // https://github.com/guillaumepotier/gettext.js#required-json-format
+  const formattedPO = {};
+  formattedPO[''] = {};
+  formattedPO[''].language = rawPO.headers.language;
+  formattedPO['']['plural-forms'] = rawPO.headers['plural-forms'];
+  delete rawPO.translations[''][''];
+  for (const msgid of Object.keys(rawPO.translations[''])) {
+    if (rawPO.translations[''][msgid].msgstr.length === 1) {
+      formattedPO[msgid] = rawPO.translations[''][msgid].msgstr[0];
+    } else {
+      formattedPO[msgid] = rawPO.translations[''][msgid].msgstr;
+    }
+  }
+  return formattedPO;
+}
+
+/** Load message catalogs for each language so they're available to i18n. */
+async function loadMessageCatalogs() {
+  // load each language's message catalog PO file into an object
+  // for easy access when we switch languages
+  fs.readdir(
+    `${__dirname}/../static/internationalization/locales`,
+    async (err, languages) => {
+      if (languages) {
+        for (const language of languages) {
+          const messageCatalogPath = `${__dirname}/../static/internationalization/locales/${language}/LC_MESSAGES/messages.po`;
+          i18n.loadJSON(readMessageCatalog(messageCatalogPath), 'messages');
+        }
+      }
+    }
+  );
+  console.log('loaded message catalogs');
+}
+loadMessageCatalogs();
+
+export default function setupSetLanguage() {
+  ipcMain.handle(
```

Comment:
I honestly have no idea what the difference is between `invoke/handle` and `send/on`, despite googling it many times. There is so much conflicting information. However I just found this [blog post](https://nornagon.medium.com/electrons-remote-module-considered-harmful-70d69500f31) from one of the electron developers. He gives an example of using `invoke/handle`, then says
> Or, using ipcRenderer.send (for Electron 6 and older):
Note that this approach can only handle a single outstanding request at a time, unless you do some bookkeeping to track which response belongs to which request. (invoke() handle matching responses to requests automatically.)

Seems to suggest that `invoke/handle` is the new preferred way. And then there's [this stackoverflow answer](https://stackoverflow.com/a/59889863/14451410), but a commenter disagrees with it. 🤷 

I used `sendSync` later in this file only because I think it needs to be synchronous, and there's no synchronous form of `invoke`.

Reply:
Okay, it sounds like there's limited use-cases for `send/on` at this point.